app/controllers/phases_controller.rb

```diff
@@ -26,4 +25,80 @@ def set_phase
     @phase = Phase.where(challenge: current_user.challenge_manager_challenges).find(params[:id])
     @challenge = @phase.challenge
   end
+
+  def set_submission_counts
+    @submissions_count = @submissions.count
+    @eligible_count = @submissions.eligible_for_evaluation.count
+    @selected_count = @submissions.winner.count
+  end
+
+  def set_submission_statuses
+    @not_started = @submissions.where.missing(:evaluations)
+    @in_progress = @submissions.joins(:evaluations).
+      where(evaluations: { completed_at: nil }).distinct
+    @completed = @submissions.joins(:evaluations).
+      where.not(evaluations: { completed_at: nil }).distinct
```

Comment:
![Screenshot 2025-01-02 at 6 40 16 PM](https://github.com/user-attachments/assets/23dc10bb-bc47-4497-9d66-503ffaf77a8f)
There is an issue in the completed query (shown in the incorrect status counts that don't add up to the total count) where it is including submissions where *any* evaluations are completed. the correct behavior should be showing submissions where *all* evaluations are completed. it took me a while to figure out, but I think this does the trick the simplest way possible:
```suggestion
    @completed = @submissions.joins(:evaluations).
      where.not(evaluations: @in_progress).distinct
```
translation: select the submissions that have at least one evaluation where none of the evaluations are in the @in_progress subquery.

Reply:
Okay interesting, thanks! I had some submissions that are recused but still assigned, so they factored into the total submissions count but not the completed/in progress/not started counts, I thought it was coming from that.